src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/Profile.java

```diff
@@ -0,0 +1,387 @@
+
+  private Profile() {
+    this.linearAvf = null;
+    this.floorAreaWeightRange = null;
+    this.nbBedroomsWeightRange = null;
+    this.nbSleepingWeightRange = null;
+    this.nbBathroomsWeightRange = null;
+    this.terraceWeightRange = null;
+    this.floorAreaTerraceWeightRange = null;
+    this.wifiSubjectiveValueWeightRange = null;
+    this.pricePerNightSubjectiveValueWeightRange = null;
+    this.nbMinNightSubjectiveValueWeightRange = null;
+    this.teleSubjectiveValueWeightRange = null;
+  }
+
+  /**
+   * This function allows the user to clone an object Profile
+   *
+   * @return an object Profile
+   */
+  private Profile cloneProfile() {
+
+    Profile prof = new Profile();
+
+    prof.linearAvf = this.linearAvf;
+    prof.floorAreaWeightRange = this.floorAreaWeightRange;
+    prof.nbBedroomsWeightRange = this.nbBedroomsWeightRange;
+    prof.nbSleepingWeightRange = this.nbSleepingWeightRange;
+    prof.nbBathroomsWeightRange = this.nbBathroomsWeightRange;
+    prof.terraceWeightRange = this.terraceWeightRange;
+    prof.floorAreaTerraceWeightRange = this.floorAreaTerraceWeightRange;
+    prof.wifiSubjectiveValueWeightRange = this.wifiSubjectiveValueWeightRange;
+    prof.pricePerNightSubjectiveValueWeightRange = this.pricePerNightSubjectiveValueWeightRange;
+    prof.nbMinNightSubjectiveValueWeightRange = this.nbMinNightSubjectiveValueWeightRange;
+    prof.teleSubjectiveValueWeightRange = this.teleSubjectiveValueWeightRange;
+
+    return prof;
+  }
+
+  /**
+   * Gives the linearAVF of a Profile
+   *
+   * @return the LinearAVF of the Profile
+   */
+  public LinearAVF getLinearAVF() {
+    return this.linearAvf;
+  }
+
+  /**
+   * Gives the subjective value weight of a criterion
+   *
+   * @param crit the criterion we want to know the value
+   * @return the subjective value weight
+   */
+  public Range<Double> getWeightRange(Criterion crit) {
+    switch (crit) {
+      case TELE:
+        return teleSubjectiveValueWeightRange;
+      case TERRACE:
+        return terraceWeightRange;
+      case WIFI:
+        return wifiSubjectiveValueWeightRange;
+      case FLOOR_AREA:
+        return floorAreaWeightRange;
+      case FLOOR_AREA_TERRACE:
+        return floorAreaTerraceWeightRange;
+      case NB_BATHROOMS:
+        return nbBathroomsWeightRange;
+      case NB_BEDROOMS:
+        return nbBedroomsWeightRange;
+      case NB_SLEEPING:
+        return nbSleepingWeightRange;
+      case NB_MIN_NIGHT:
+        return nbMinNightSubjectiveValueWeightRange;
+      case PRICE_PER_NIGHT:
+        return pricePerNightSubjectiveValueWeightRange;
+      default:
+        throw new IllegalArgumentException();
+    }
+  }
+
+  /**
+   * Get the middle of the given Range
+   *
+   * @param range the Range
+   * @return the middle of the given Range of Doubles
+   */
+  static double getMiddleOfRange(Range<Double> range) {
+    checkNotNull(range);
+    return (range.upperEndpoint() + range.lowerEndpoint()) / 2;
+  }
```

Comment:
Pas convaincu par l’utilité de cette méthode d’une ligne. OPT. (Je trouve que la suivante, `getMiddleOfRange(crit)`, passe la barre tout juste car elle est un poil plus compliquée conceptuellement et requiert d’accéder aux donnés de cet objet.)

Reply:
Fixed in commit 6f117ca